src/ReplaySession.cc

```diff
@@ -1042,25 +1043,29 @@ Completion ReplaySession::emulate_async_signal(
         // breakpoint instruction in the tracee would have triggered a
         // deterministic signal instead of an async one.
         // So we must have hit our internal breakpoint.
-        ASSERT(t, did_set_internal_breakpoints);
+        ASSERT(t, did_set_internal_breakpoints || did_set_bpf_breakpoint);
```

Comment:
Would it be better to always use a hardware perf-event breakpoint and just not use the BPF program when BPF isn't available? I think that would be a bit simpler?

Reply:
Permanently consuming a hardware breakpoint for people who aren't even getting the benefit of bpf seems bad.

Reply:
Yeah, that makes sense.